quacc/calculators/qchem.py

```diff
@@ -0,0 +1,147 @@
+        self.cores = cores
+        self.charge = charge
+        self.spin_multiplicity = spin_multiplicity
+        self.qchem_input_params = qchem_input_params
+        self.use_custodian = use_custodian
+        self.qchem_input_params = qchem_input_params or {}
+        self.kwargs = kwargs
+
+        # Instantiate the calculator
+        FileIOCalculator.__init__(
+            self,
+            restart=None,
+            ignore_bad_restart_file=FileIOCalculator._deprecated,
+            label=None,
+            atoms=self.input_atoms,
+            **self.kwargs,
+        )
+
+        # Get Q-Chem executable command
+        command = self._manage_environment()
+
+    def _manage_environment(self) -> str:
+        """
+        Manage the environment for the Q-Chem calculator.
+
+        Returns
+        -------
+        str
+            The command flag to pass to the Q-Chem calculator.
+        """
+
+        # Check if Custodian should be used
+        if self.use_custodian:
+            # Return the command flag
+            run_qchem_custodian_file = os.path.abspath(inspect.getfile(custodian_qchem))
+            return f"python {run_qchem_custodian_file} {self.cores}"
+        else:
+            warnings.warn("Can only run Q-Chem via Custodian currently!")
+            return None
+
+    def write_input(self, atoms, properties=None, system_changes=None):
+        FileIOCalculator.write_input(self, atoms, properties, system_changes)
+        mol = AseAtomsAdaptor.get_molecule(atoms)
+        if self.charge is not None:
+            if self.spin_multiplicity is not None:
+                mol.set_charge_and_spin(
+                    charge=self.charge, spin_multiplicity=self.spin_multiplicity
+                )
+            else:
+                mol.set_charge_and_spin(charge=self.charge)
+        qcin = ForceSet(mol, **self.qchem_input_params)
+        qcin.write("mol.qin")
+
+    def read_results(self):
+        data = QCOutput("mol.qout").data
+        self.results["energy"] = data["final_energy"] * units.Hartree
+        tmp_grad_data = []
+        with zopen("131.0", mode="rb") as file:
+            binary = file.read()
+            for ii in range(int(len(binary) / 8)):
+                tmp_grad_data.append(
+                    struct.unpack("d", binary[ii * 8 : (ii + 1) * 8])[0]
+                )
+        grad = []
+        for ii in range(int(len(tmp_grad_data) / 3)):
+            grad.append(
+                [
+                    float(tmp_grad_data[ii * 3]),
+                    float(tmp_grad_data[ii * 3 + 1]),
+                    float(tmp_grad_data[ii * 3 + 2]),
+                ]
+            )
+        if data["pcm_gradients"] is not None:
+            gradient = data["pcm_gradients"][0]
+        else:
+            gradient = data["gradients"][0]
+        for ii, subgrad in enumerate(grad):
+            for jj, val in enumerate(subgrad):
+                assert abs(gradient[ii, jj] - val) < 1e-6
```

Comment:
`assert` should only be used in test suites. Use an `if` statement here and raise a `ValueError` if there's an issue.

Reply:
Fixed